### PR TITLE
Add ability to delete via admin API

### PR DESCRIPTION
**Pull request by bjacobso**

### CODE REVIEW

File: admin.go [hunk 1]
```diff
@@ -55,6 +56,34 @@ func (om *OpenMock) StartAdmin() {
 		return c.String(200, string(b))
 	})
 
+	e.DELETE("/api/v1/templates/:key", func(c echo.Context) error {
+		key, err := url.QueryUnescape(c.Param("key"))
+		if err != nil {
+			return err
```
Comment (zhouzhuojie) [inline admin.go]: can this be more specific? for example, returning 400

```
return c.String(400, fmt.Sprintf("invalid Key: %v. error: %s", key, err))

```

**Comment by oriahu** [comment]
This also deletes vendor directory, is that something is gonna change? I think we decided on keeping vendor directory to not re-pull everything each time.  though i could be wrong

**Comment by zhouzhuojie** [comment]
> This also deletes vendor directory, is that something is gonna change? I think we decided on keeping vendor directory to not re-pull everything each time. though i could be wrong

I think the vendor folder was just updated (with some deletion), not removed.

File: admin.go [hunk 2]
```diff
@@ -55,6 +56,34 @@ func (om *OpenMock) StartAdmin() {
 		return c.String(200, string(b))
 	})
 
+	e.DELETE("/api/v1/templates/:key", func(c echo.Context) error {
+		key, err := url.QueryUnescape(c.Param("key"))
+		if err != nil {
+			return c.String(400, fmt.Sprintf("Invalid Key: %v. error: %s", key, err))
+		}
+
+		mock := om.repo.FindByKey(key)
```
Comment (zhouzhuojie) [inline admin.go]: nit, do you need this? I think you can just try HGET in redis, since the delete only applies to redis storage.

Comment (bjacobso) [reply admin.go]: Updated to check redis and return the deleted yaml if its successfully deleted

**Pull request merged**